File: solutions/cpp/log-levels/log_levels.cpp
#include <string>

namespace log_line {
    const std::string delimiter = ": ";

    std::string message(std::string line) {
        return line.substr(line.find(delimiter) + delimiter.length());
    }

    std::string log_level(std::string line) {
        return line.substr(1, line.find(delimiter) - 2);
    }

    std::string reformat(std::string line) {
        return message(line) + " (" + log_level(line) + ")";
    }
} // namespace log_line

File: solutions/java/log-levels/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
        var startPos = logLine.indexOf(": ") + 1;
        return logLine.substring(startPos).trim();
    }

    public static String logLevel(String logLine) {
        var endPos = logLine.indexOf("]:");
        return logLine.substring(1, endPos).toLowerCase();
    }

    public static String reformat(String logLine) {
        return message(logLine) + " (" + logLevel(logLine) + ")";
    }
}
